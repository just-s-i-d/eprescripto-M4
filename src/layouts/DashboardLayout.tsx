import { SetStateAction, useContext, useState } from "react";
import { Outlet, useLocation } from "react-router-dom";

import { Breadcrumb, Layout, theme, Space } from "antd";
import Item from "antd/es/list/Item";

import NavBar from "@components/NavBar/NavBar";
import SideBar from "@components/SideBar/Sidebar";
import { UserContext } from "@context/UserProvider";
import Loading from "@components/ui/Loading";

type Props = {
  setIsDarkTheme: React.Dispatch<SetStateAction<boolean>>;
};

const DashboardLayout: React.FC<Props> = ({ setIsDarkTheme }) => {
  const userContext = useContext(UserContext);
  const location = useLocation();
  const [collapsed, setCollapsed] = useState<boolean>(false);
  const {
    token: { colorText, colorTextSecondary, colorPrimaryBg },
  } = theme.useToken();

  if (userContext === null) return <Loading />;
  const { role } = userContext;

  return (
    <Layout className="min-h-screen h-full relative">
      <NavBar
        collapsed={collapsed}
        setCollapsed={setCollapsed}
        colorPrimaryBg={colorPrimaryBg}
        colorText={colorText}
      />
      <Layout className="relative">
        <SideBar
          role={role}
          textColor={colorTextSecondary}
          setIsDarkTheme={setIsDarkTheme}
          collapsed={collapsed}
          setCollapsed={setCollapsed}
        />
        <Layout className="pb-6 px-6">
          <Space className="flex justify-between" direction="horizontal">
            <Breadcrumb className="text-2xl my-4">
              <Item>{location.pathname.split("/")[1].toUpperCase()}</Item>
            </Breadcrumb>
          </Space>
          <Outlet />
        </Layout>
      </Layout>
    </Layout>
  );
};

export default DashboardLayout;
