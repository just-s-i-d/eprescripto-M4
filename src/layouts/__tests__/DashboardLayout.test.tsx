import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { MemoryRouter } from "react-router-dom";
import "@test/matchMedia";
import DashboardLayout from "../DashboardLayout";

describe("DashboardLayout", () => {
  it("Test to check if NavBar and Sidebar is rendered", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <DashboardLayout setIsDarkTheme={jest.fn()} />
      </MemoryRouter>,
    );
    expect(screen.getByText("ePrescripto")).toBeInTheDocument();
    expect(screen.getByRole("switch")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
