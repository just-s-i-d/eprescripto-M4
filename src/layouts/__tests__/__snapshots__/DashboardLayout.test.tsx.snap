// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardLayout Snapshot test 1`] = `
<DocumentFragment>
  <div
    class="ant-layout min-h-screen h-screen relative overflow-hidden css-dev-only-do-not-override-gzal6t"
  >
    <header
      class="ant-layout-header px-6 sticky top-0 z-10 flex items-center justify-between h-[8vh] css-dev-only-do-not-override-gzal6t"
    >
      <span
        class="text-3xl text-white"
      >
        <button
          class="ant-btn css-dev-only-do-not-override-gzal6t ant-btn-text ant-btn-icon-only text-sm text-white w-8 h-8"
          data-testid="side-menu-btn"
          type="button"
        >
          <span
            class="ant-btn-icon"
          >
            <span
              aria-label="menu-fold"
              class="anticon anticon-menu-fold"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="menu-fold"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM115.4 518.9L271.7 642c5.8 4.6 14.4.5 14.4-6.9V388.9c0-7.4-8.5-11.5-14.4-6.9L115.4 505.1a8.74 8.74 0 000 13.8z"
                />
              </svg>
            </span>
          </span>
        </button>
        <a
          href="/"
        >
          ePrescripto
        </a>
      </span>
      <div
        class="ant-space css-dev-only-do-not-override-gzal6t ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
      >
        <div
          class="ant-space-item"
        >
          <span
            class="ant-badge css-dev-only-do-not-override-gzal6t"
          >
            <span
              class="ant-avatar ant-avatar-circle ant-avatar-icon css-dev-only-do-not-override-gzal6t"
              style="width: 42px; height: 42px; line-height: 42px; font-size: 21px;"
            >
              <span
                aria-label="user"
                class="anticon anticon-user"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="user"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
                  />
                </svg>
              </span>
            </span>
          </span>
        </div>
      </div>
    </header>
    <div
      class="ant-layout ant-layout-has-sider flex flex-row css-dev-only-do-not-override-gzal6t"
    >
      <aside
        class="ant-layout-sider ant-layout-sider-dark h-[92vh] max-xl:top-[8vh] left-0 max-xl:sticky max-lg:fixed z-10"
        style="flex: 0 0 250px; max-width: 250px; min-width: 250px; width: 250px;"
      >
        <div
          class="ant-layout-sider-children"
        >
          <ul
            class="ant-menu ant-menu-root ant-menu-inline ant-menu-light h-full text-xl px-2 relative css-dev-only-do-not-override-gzal6t"
            data-menu-list="true"
            role="menu"
            tabindex="0"
          >
            <li
              class="ant-menu-item my-6 py-2 flex"
              data-menu-id="rc-menu-uuid-test-dashboard"
              role="menuitem"
              style="padding-left: 24px;"
              tabindex="-1"
            >
              <span
                aria-label="home"
                class="anticon anticon-home ant-menu-item-icon"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="home"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M946.5 505L560.1 118.8l-25.9-25.9a31.5 31.5 0 00-44.4 0L77.5 505a63.9 63.9 0 00-18.8 46c.4 35.2 29.7 63.3 64.9 63.3h42.5V940h691.8V614.3h43.4c17.1 0 33.2-6.7 45.3-18.8a63.6 63.6 0 0018.7-45.3c0-17-6.7-33.1-18.8-45.2zM568 868H456V664h112v204zm217.9-325.7V868H632V640c0-22.1-17.9-40-40-40H432c-22.1 0-40 17.9-40 40v228H238.1V542.3h-96l370-369.7 23.1 23.1L882 542.3h-96.1z"
                  />
                </svg>
              </span>
              <span
                class="ant-menu-title-content"
              >
                <a
                  class="text-lg"
                  href="/dashboard"
                >
                  Dashboard
                </a>
              </span>
            </li>
            <li
              class="ant-menu-item my-6 py-2 flex"
              data-menu-id="rc-menu-uuid-test-prescriptions"
              role="menuitem"
              style="padding-left: 24px;"
              tabindex="-1"
            >
              <span
                aria-label="form"
                class="anticon anticon-form ant-menu-item-icon"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="form"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M904 512h-56c-4.4 0-8 3.6-8 8v320H184V184h320c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V520c0-4.4-3.6-8-8-8z"
                  />
                  <path
                    d="M355.9 534.9L354 653.8c-.1 8.9 7.1 16.2 16 16.2h.4l118-2.9c2-.1 4-.9 5.4-2.3l415.9-415c3.1-3.1 3.1-8.2 0-11.3L785.4 114.3c-1.6-1.6-3.6-2.3-5.7-2.3s-4.1.8-5.7 2.3l-415.8 415a8.3 8.3 0 00-2.3 5.6zm63.5 23.6L779.7 199l45.2 45.1-360.5 359.7-45.7 1.1.7-46.4z"
                  />
                </svg>
              </span>
              <span
                class="ant-menu-title-content"
              >
                <a
                  class="text-lg"
                  href="/prescriptions"
                >
                  Prescriptions
                </a>
              </span>
            </li>
            <li
              class="ant-menu-item my-6 py-2 flex"
              data-menu-id="rc-menu-uuid-test-appointments"
              role="menuitem"
              style="padding-left: 24px;"
              tabindex="-1"
            >
              <span
                aria-label="schedule"
                class="anticon anticon-schedule ant-menu-item-icon"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="schedule"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M928 224H768v-56c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v56H548v-56c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v56H328v-56c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v56H96c-17.7 0-32 14.3-32 32v576c0 17.7 14.3 32 32 32h832c17.7 0 32-14.3 32-32V256c0-17.7-14.3-32-32-32zm-40 568H136V296h120v56c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-56h148v56c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-56h148v56c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-56h120v496zM416 496H232c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h184c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8zm0 136H232c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h184c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8zm308.2-177.4L620.6 598.3l-52.8-73.1c-3-4.2-7.8-6.6-12.9-6.6H500c-6.5 0-10.3 7.4-6.5 12.7l114.1 158.2a15.9 15.9 0 0025.8 0l165-228.7c3.8-5.3 0-12.7-6.5-12.7H737c-5-.1-9.8 2.4-12.8 6.5z"
                  />
                </svg>
              </span>
              <span
                class="ant-menu-title-content"
              >
                <a
                  class="text-lg"
                  href="/appointments"
                >
                  Appointments
                </a>
              </span>
            </li>
            <li
              class="ant-menu-item my-6 py-2 flex"
              data-menu-id="rc-menu-uuid-test-doctors"
              role="menuitem"
              style="padding-left: 24px;"
              tabindex="-1"
            >
              <span
                aria-label="star"
                class="anticon anticon-star ant-menu-item-icon"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="star"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3zM664.8 561.6l36.1 210.3L512 672.7 323.1 772l36.1-210.3-152.8-149L417.6 382 512 190.7 606.4 382l211.2 30.7-152.8 148.9z"
                  />
                </svg>
              </span>
              <span
                class="ant-menu-title-content"
              >
                <a
                  class="text-lg"
                  href="/doctors"
                >
                  Doctors
                </a>
              </span>
            </li>
            <li
              class="ant-menu-item my-6 py-2 flex"
              data-menu-id="rc-menu-uuid-test-profile"
              role="menuitem"
              style="padding-left: 24px;"
              tabindex="-1"
            >
              <span
                aria-label="profile"
                class="anticon anticon-profile ant-menu-item-icon"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="profile"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-40 728H184V184h656v656zM492 400h184c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8H492c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8zm0 144h184c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8H492c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8zm0 144h184c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8H492c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8zM340 368a40 40 0 1080 0 40 40 0 10-80 0zm0 144a40 40 0 1080 0 40 40 0 10-80 0zm0 144a40 40 0 1080 0 40 40 0 10-80 0z"
                  />
                </svg>
              </span>
              <span
                class="ant-menu-title-content"
              >
                <a
                  class="text-lg"
                  href="/profile"
                >
                  Profile
                </a>
              </span>
            </li>
            <button
              aria-checked="false"
              class="ant-switch absolute bottom-6 css-dev-only-do-not-override-gzal6t"
              eventkey="tmp_key-5"
              role="switch"
              style="background-color: rgb(230, 244, 255);"
              type="button"
            >
              <div
                class="ant-switch-handle"
              />
              <span
                class="ant-switch-inner"
              >
                <span
                  class="ant-switch-inner-checked"
                >
                  Dark
                </span>
                <span
                  class="ant-switch-inner-unchecked"
                >
                  Light
                </span>
              </span>
            </button>
          </ul>
          <div
            aria-hidden="true"
            style="display: none;"
          >
            <button
              aria-checked="false"
              class="ant-switch absolute bottom-6 css-dev-only-do-not-override-gzal6t"
              eventkey="tmp_key-5"
              role="switch"
              style="background-color: rgb(230, 244, 255);"
              type="button"
            >
              <div
                class="ant-switch-handle"
              />
              <span
                class="ant-switch-inner"
              >
                <span
                  class="ant-switch-inner-checked"
                >
                  Dark
                </span>
                <span
                  class="ant-switch-inner-unchecked"
                >
                  Light
                </span>
              </span>
            </button>
          </div>
        </div>
      </aside>
      <div
        class="ant-layout pb-6 px-6 max-sm:px-3 overflow-y-scroll overflow-x-hidden css-dev-only-do-not-override-gzal6t"
      >
        <div
          class="ant-space css-dev-only-do-not-override-gzal6t ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small flex justify-between"
        >
          <div
            class="ant-space-item"
          >
            <nav
              class="ant-breadcrumb text-2xl my-4 css-dev-only-do-not-override-gzal6t"
            >
              <ol>
                <li>
                  <span
                    class="ant-breadcrumb-link"
                  />
                </li>
              </ol>
            </nav>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
